File: app-laravel/src/Product/Application/Commands/UpdateStock/UpdateStockCommand.php
<?php

declare(strict_types=1);

namespace Src\Product\Application\Commands\UpdateStock;

use Src\Shared\Application\Bus\Command;

/**
 * UpdateStockCommand - Məhsul stokunu yeniləmə əmri.
 *
 * Bu command stoku artırmaq və ya azaltmaq üçün istifadə olunur.
 * "type" sahəsi əməliyyat növünü müəyyən edir: 'increase' və ya 'decrease'.
 */
final class UpdateStockCommand implements Command
{
    /**
     * @param string $productId Məhsulun ID-si
     * @param int    $amount    Dəyişiklik miqdarı (müsbət rəqəm)
     * @param string $type      Əməliyyat növü: 'increase' (artır) və ya 'decrease' (azalt)
     */
    public function __construct(
        public readonly string $productId,
        public readonly int $amount,
        public readonly string $type, // 'increase' və ya 'decrease'
    ) {
    }
}

File: app/spring/src/main/java/az/ecommerce/product/application/command/updatestock/UpdateStockCommand.java
package az.ecommerce.product.application.command.updatestock;

import az.ecommerce.shared.application.bus.Command;
import jakarta.validation.constraints.*;

import java.util.UUID;

public record UpdateStockCommand(
        @NotNull UUID productId,
        @Positive int amount,
        @NotBlank @Pattern(regexp = "increase|decrease") String type
) implements Command<Void> {}
